EXILED/Exiled.API/Features/Player.cs

```diff
@@ -2835,6 +2841,8 @@ public Item AddItem(ItemBase itemBase, Item item = null, ItemAddReason addReason
                     acquisitionConfirmationTrigger.AcquisitionAlreadyReceived = false;
                 }
 
+                itemBase.ServerAddReason = addReason;
+                itemBase.OnAdded(null);
```

Comment:
That `OnAdded` will broke many things like microhid or radio, for items that should have specific logic for adding/removing there are method inside item class `ChangeOwner`. (i suppose now we should call `OnAdded` for firearm for changing owner, yes?)

Reply:
No, we call `OnAdded` because that's where it's submodules are initialized when the item is created. It's not related to `ChangeOwner` at all.
There may be some way to determine if the item *should* have this called, but base-game `ServerAddItem()` always calls it, so I replicated it here since `AddItem()` is supposed to be the API replacement for that basegame method.

Reply:
We could extract just the submodule initialization itself from `ServerAddItem` here instead, though I'm unsure what the current method could do to break something.

Reply:
Yes, but basegame not support removing/added item *manually*, from inventory to inventory, without pickups or projectiles.
Logical meaning of `ServerAddItem` to add `itemtype` not `ItemBase`, so ofc it will have `OnAdded` it literally create item in it.

But exiled method is different, it editing `ItemBase`, and moves it to another player, and not for all items in that case we should call `OnAdded`

Reply:
Then we'll need to extract the submodule initialization out and add something to determine if we need to call it or not.

Reply:
Exiled AddItem is a some part of abstraction, so it works on a higher level than `ServerAddItem`

Reply:
> Then we'll need to extract the submodule initialization out and add something to determine if we need to call it or not.

That it why `ChangeOwner` methods exist, they called only for that cases

Reply:
> Then we'll need to extract the submodule initialization out and add something to determine if we need to call it or not.

Actually, how do you add items?

Bcz `ply.AddItem(Item.Create(...))` should have no problems
`OnAdded` calling in item ctor

Reply:
I didn't test using item create, I'm using the ItemType overload for testing.